tests/Makefile

```diff
@@ -16,59 +17,21 @@ OS ?= $(HOSTOS)
 
 WORKDIR ?= $(CURDIR)/../dist
 
-test: units_test integration_test lim_test docker_test
+test: $(TESTS:=_test)
 
-build: integration_build reboot_build lim_build docker_build
-setup: integration_setup reboot_setup lim_setup docker_setup
-clean: integration_clean reboot_clean lim_clean docker_clean
+build: $(TESTS:=_build)
+setup: $(TESTS:=_setup)
+clean: $(TESTS:=_clean)
 
-units_test:
-	make -C units test
+%_test: % %_build %_setup
+	make -C $< DEBUG=$(DEBUG) ARCH=$(ARCH) OS=$(OS) WORKDIR=$(WORKDIR) test
 
-integration_test: integration_build integration_setup
-	make -C integration DEBUG=$(DEBUG) ARCH=$(ARCH) OS=$(OS) WORKDIR=$(WORKDIR) test
+%_build: %
+	make -C $< DEBUG=$(DEBUG) ARCH=$(ARCH) OS=$(OS) WORKDIR=$(WORKDIR) build
 
-reboot_test: reboot_build reboot_setup
-	make -C reboot DEBUG=$(DEBUG) ARCH=$(ARCH) OS=$(OS) WORKDIR=$(WORKDIR) test
+%_setup: %
+	make -C $< DEBUG=$(DEBUG) ARCH=$(ARCH) OS=$(OS) WORKDIR=$(WORKDIR) setup
 
-lim_test: lim_build lim_setup
-	make -C lim DEBUG=$(DEBUG) ARCH=$(ARCH) OS=$(OS) WORKDIR=$(WORKDIR) test
+%_clean: %
+	make -C $< DEBUG=$(DEBUG) ARCH=$(ARCH) OS=$(OS) WORKDIR=$(WORKDIR) clean
 
-docker_test: docker_build docker_setup
```

Comment:
docker test? Or we will add later?

Reply:
No. The main change of this refactoring -- automatization of tests list generation.Now the developer should simply add his own test folder to the tests/ directory, and test will automagically build/run/clean without having to change the top-level Makefiles.